File: ReactAndroid/src/main/java/com/facebook/yoga/YogaLogger.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.yoga;

import androidx.annotation.Keep;

/**
 * Interface for receiving logs from native layer. Use by setting YogaNode.setLogger(myLogger);
 * See YogaLogLevel for the different log levels.
 */
@Keep
public interface YogaLogger {
  @Keep
  void log(YogaLogLevel level, String message);
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/yoga/YogaLogger.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.yoga

import com.facebook.yoga.annotations.DoNotStrip

/**
 * Interface for receiving logs from native layer. Use by setting YogaNode.setLogger(myLogger); See
 * YogaLogLevel for the different log levels.
 */
@DoNotStrip
public fun interface YogaLogger {
  @DoNotStrip public fun log(level: YogaLogLevel, message: String)
}
